### PR TITLE
release-manifests: allow overriding fetch implementation

### PR DESCRIPTION
**Pull request by mtlewis**
## Hey, I just made a Pull Request!

This allows us to use yarn's built in HTTP request utilities in the Backstage yarn plugin, thereby benefiting from the built-in caching and proxy configuration utilities.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/release-manifests/report.api.md [hunk 1]
```diff
@@ -21,6 +21,12 @@ export function getManifestByVersion(
 // @public
 export type GetManifestByVersionOptions = {
   version: string;
+  fetchFn?: (
+    url: string,
+    options?: {
+      signal?: AbortSignal;
+    },
+  ) => Promise<Pick<Response, 'status' | 'json' | 'url'>>;
```
Comment (mtlewis) [inline packages/release-manifests/report.api.md:29]: Just flagging that we're potentially risking breaking changes here if in the future we need more of the properties of `Response` after fetching.

File: packages/release-manifests/report.api.md [hunk 2]
```diff
@@ -21,6 +21,12 @@ export function getManifestByVersion(
 // @public
 export type GetManifestByVersionOptions = {
   version: string;
+  fetchFn?: (
```
Comment (Rugvip) [inline packages/release-manifests/report.api.md:24]:
```suggestion
  fetch?: (
```

since fetch is already a function I'm thinking that's enough

Comment (mtlewis) [reply packages/release-manifests/report.api.md]: Updated :+1:

**Review by Rugvip — APPROVED** [review]
:shipit:

**Pull request merged**